vector/src/main/java/im/vector/app/features/html/EventHtmlRenderer.kt

```diff
@@ -20,24 +20,51 @@ import android.content.Context
 import android.text.Spannable
 import androidx.core.text.toSpannable
 import im.vector.app.core.resources.ColorProvider
+import im.vector.app.features.settings.VectorPreferences
+import io.noties.markwon.AbstractMarkwonPlugin
 import io.noties.markwon.Markwon
+import io.noties.markwon.MarkwonPlugin
+import io.noties.markwon.PrecomputedFutureTextSetterCompat
+import io.noties.markwon.ext.latex.JLatexMathPlugin
+import io.noties.markwon.ext.latex.JLatexMathTheme
 import io.noties.markwon.html.HtmlPlugin
+import io.noties.markwon.inlineparser.MarkwonInlineParserPlugin
 import org.commonmark.node.Node
 import timber.log.Timber
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class EventHtmlRenderer @Inject constructor(htmlConfigure: MatrixHtmlPluginConfigure,
-                                            context: Context) {
+                                            context: Context,
+                                            private val vectorPreferences: VectorPreferences) {
 
     interface PostProcessor {
         fun afterRender(renderedText: Spannable)
     }
 
-    private val markwon = Markwon.builder(context)
+    private val builder = Markwon.builder(context)
             .usePlugin(HtmlPlugin.create(htmlConfigure))
-            .build()
+
+    private val markwon = if (vectorPreferences.latexMathsIsEnabled()) {
+        builder
+                .usePlugin(object : AbstractMarkwonPlugin() { // Markwon expects maths to be in a specific format: https://noties.io/Markwon/docs/v4/ext-latex
+                    override fun processMarkdown(markdown: String): String {
+                        return markdown
+                                .replace(Regex("""<span\s+data-mx-maths="([^"]*)">.*?</span>""")) { matchResult -> "$$" + matchResult.groupValues[1] + "$$" }
+                                .replace(Regex("""<div\s+data-mx-maths="([^"]*)">.*?</div>""")) { matchResult -> "\n$$\n" + matchResult.groupValues[1] + "\n$$\n" }
+                    }
+                })
+                .usePlugin(MarkwonInlineParserPlugin.create())
+                .usePlugin(JLatexMathPlugin.create(44F) { builder ->
+                    builder.inlinesEnabled(true)
+                    builder.theme().inlinePadding(JLatexMathTheme.Padding.symmetric(24, 8))
+                })
+    } else {
+        builder
+    }.textSetter(PrecomputedFutureTextSetterCompat.create()).build()
```

Comment:
What is the goal of `textSetter(PrecomputedFutureTextSetterCompat.create())`?

Reply:
See https://github.com/noties/Markwon/issues/218. Element uses AppCompatTextView everywhere anyway. I thought it might help with the clipping issue, but it didn't. Doesn't hurt to use this anyway.